cmd/txtar-x/extract_test.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/rogpeppe/go-internal/testscript"
+)
+
+func TestMain(m *testing.M) {
+	os.Exit(testscript.RunMain(m, map[string]func() int{
+		"txtar-x": main1,
+	}))
+}
+
+func TestScripts(t *testing.T) {
+	testscript.Run(t, testscript.Params{
+		Dir: "testdata",
+		Cmds: map[string]func(ts *testscript.TestScript, neg bool, args []string){
+			"unquote": unquote,
+		},
+	})
+}
+
+func unquote(ts *testscript.TestScript, neg bool, args []string) {
```

Comment:
Nice approach. An alternative would be extend the `txtar` filename format in some way to indicate that the contents should be unquoted, e.g. 

```
-- > file.txtar --
>some comment
>
>-- foo --
>foo
>-- a/b/bar --
>bar
```

Reply:
meh. I like the simplicity of the current syntax (no errors possible), and if it can be worked around this easily, then I don't there's a particular need to change it.